### ISSUE TITLE
"routing not found for "undefined""

### ISSUE THREAD

**Issue by otiai10** [otiai10/chomex#10]
weird

**Comment by otiai10** [comment]
- NOT always
- Controllers are executed

**Comment by otiai10** [comment]
```
Uncaught (in promise) Error: Attempting to use a disconnected port object
    at PortImpl.postMessage (extensions::messaging:60:13)
    at Port.publicClassPrototype.(anonymous function) [as postMessage] (extensions::utils:149:26)
    at responseCallback (extensions::messaging:166:16)
    at chrome-extension://baeembooipohpmanjnadeeklaajpgfmj/dest/js/background.js:774:21
```

TODO: cause or effect?

**Comment by otiai10** [comment]
facts:

- client: `404` `routing not found for "undefined"`
- server: `Uncaught (in promise) Error: Attempting to use a disconnected port object`

**Comment by otiai10** [comment]
# tl;dr

it was because of 2 problems

- **Client**: Send message without params, like `client.message('/status/check')`, didn't give parameters object correctly to server.
    - fixed by https://github.com/otiai10/chomex/commit/36bfdc51c3ecfb1ce65454ec98c336946e8b522f
- **Server**: Tried to sendResponse several minutes later, even that connection was already closed.
    - when the response might be late, sendResponse first and send specific message to tab.
        - e.g. https://github.com/otiai10/kanColleWidget/commit/0fcf94129d00dc10d8f2baf8a702c808b58bee5b
